hastexo/tasks.py

```diff
@@ -135,6 +135,57 @@ def run(self, **kwargs):
         for key, value in kwargs.items():
             setattr(self, key, value)
 
+        # If a time limit is set for using labs,
+        # check how much time learner has already spent
+        settings = get_xblock_settings()
+        lab_usage_limit = settings.get("lab_usage_limit", None)
+
+        policy_warn_message = None
+        if lab_usage_limit:
+            # don't look further than 365 days for lab launch
+            cutoff = timezone.now() - timezone.timedelta(days=365)
```

Comment:
Future implementation note, no action needed now we probably want to make this window configurable as `lab_usage_limit_window`.

Reply:
Ack.